CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -342,20 +342,14 @@ function Patient:falling()
   local current = self:getCurrentAction()
   current.keep_reserved = true
   if self.falling_anim and self:canPeeOrPuke(current) and self.has_fallen == 1 then
-    self:setNextAction(FallingAction(), 0)
     self.has_fallen = 2
-    if self.has_fallen == 2 then
-      self:setNextAction(OnGroundAction())
-      self.on_ground = true
-    end
-    if self.on_ground then
-      self:setNextAction(GetUpAction())
-    end
-    -- TODO: falls are broken, the will change once implemented
-    self:interruptAndRequeueAction(current, 2)
-    self.on_ground = false
-    if math.random(1, 5) == 3 then
-      self:shakeFist()
+    self:queueAction(FallingAction(), 1)
+    self:queueAction(OnGroundAction(), 2)
+    self:queueAction(GetUpAction(), 3)
+    if math.random(1, 5) == 3 and self:shakeFist() then
+      self:interruptAndRequeueAction(current, 5)
+    else
+      self:interruptAndRequeueAction(current, 4)
     end
     self:fallingAnnounce()
```

Comment:
Ah, the problem I identified a while back sits here. This ends up spamming the announcer with every person falling.
It's out of scope so just needs noting to be fixed (or fixed here, if you wanted to).

Reply:
In fact, the falling Adviser messages are irrelevant to earthquakes.... They're when you click a patient I believe to knock them over.